File: plugboard.go
package main

type Plugboard struct {
	in  string
	out string
}

func (pb *Plugboard) Map(letter string) string {
	for i, l := range pb.in {
		if string(l) == letter {
			return string(pb.out[i])
		}
	}
	for i, l := range pb.out {
		if string(l) == letter {
			return string(pb.in[i])
		}
	}
	return letter
}

File: python/plugboard.py
class Plugboard:
  def __init__(self, alphabet, output):
    self.input = alphabet
    self.output = output

  def map(self, letter):
    for i, c in enumerate(self.input):
      if c == letter:
        return self.output[i]
      if self.output[i] == letter:
        return c
    return letter